### ISSUE TITLE
vue/custom-event-name-casing not working for <script setup>

### ISSUE THREAD

**Issue by Fire7** [vuejs/eslint-plugin-vue#1697]
vue/no-unregistered-components not working for <script setup>

**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 7.21.0
- **eslint-plugin-vue version:** 8.0.3
- **Node version:** v14.17.3
- **Operating System:** MacOS

**Please show your full configuration:**

```json5
module.exports = {
  settings: {
    'import/core-modules': ['vue', 'nuxt3'], // these modules are included in nuxt.js
  },
  env: {
    browser: true,
    es6: true,
    node: true,
    'vue/setup-compiler-macros': true,
  },
  extends: [
    '@nuxtjs/eslint-config-typescript',
    'plugin:@typescript-eslint/recommended',
    'plugin:vue/vue3-recommended',
    'airbnb-base',
    'plugin:import/typescript',
  ],
  parser: 'vue-eslint-parser',
  parserOptions: {
    ecmaVersion: 2021,
    parser: '@typescript-eslint/parser',
    sourceType: 'module',
    extraFileExtensions: ['.vue'],
  },
  plugins: [
    'vue',
    '@typescript-eslint',
  ],
  rules: {
    semi: ['error', 'never'],
    'max-len': ['error', 100],
    'no-undef': 'off',
    'no-console': ['error', { allow: ['warn', 'error', 'info', 'table'] }],
    'global-require': 'off',
    'no-unused-vars': 'off',
    'consistent-return': 'off',
    'no-param-reassign': ['error', { props: false }],
    'no-use-before-define': 'off',
    'object-curly-newline': ['error', {
      minProperties: 5,
      multiline: true,
      consistent: true,
    }],
    'no-multi-assign': 'off',
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
    'import/prefer-default-export': 'off',
    'no-prototype-builtins': 'off',
    '@typescript-eslint/triple-slash-reference': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    '@typescript-eslint/naming-convention': [
      'error',
      {
        selector: 'interface',
        format: ['PascalCase'],
        custom: {
          regex: '^I[A-Z]',
          match: false,
        },
      },
      {
        selector: 'typeLike',
        format: ['PascalCase'],
      },
    ],
    'vue/attribute-hyphenation': ['error', 'never', {
      ignore: [],
    }],
    'vue/multi-word-component-names': 'off',
    'vue/prop-name-casing': ['error', 'camelCase'],
    'vue/custom-event-name-casing': ['error', 'camelCase'],
    'vue/padding-line-between-blocks': ['warn', 'always'],
    'vue/no-mutating-props': 'off',
    'vue/no-unused-properties': ['error', { groups: ['setup', 'data', 'methods'] }],
    'vue/component-name-in-template-casing': ['error', 'PascalCase'],
    'vue/no-multiple-objects-in-class': ['warn'],
    'vue/no-duplicate-attr-inheritance': ['error'],
    'vue/match-component-file-name': ['error', {
      extensions: ['.tsx', '.vue'],
      shouldMatchCase: true,
    }],
    'vue/require-name-property': ['error'],
    'vue/v-on-function-call': ['warn', 'never'],
  },
}
```

**What did you do?**	

```vue
<script setup> 
import { defineEmit } from 'vue'  

const emit = defineEmit(['snake_case', 'PascalCase'])  

const handleClick = () => {   emit('snake_case')   emit('PascalCase') } 
</script>
```

**What did you expect to happen?**
Eslint reports:
Custom event name 'PascalCase' must be camelCase        vue/custom-event-name-casing
Custom event name 'snake_case' must be camelCase        vue/custom-event-name-casing

**What actually happened?**

No lint errors found

**Repository to reproduce this issue**

**Comment by ota-meshi** [comment]
You need to use `defineEmits`. The old API `defineEmit` is not supported.

https://v3.vuejs.org/api/sfc-script-setup.html#defineprops-and-defineemits

**Comment by Fire7** [comment]
Thank you
